Objects/call.c

```diff
@@ -309,6 +309,31 @@ _PyObject_FastCall(PyObject *func, PyObject *const *args, Py_ssize_t nargs)
 }
 
 
+static PyObject*
+copy_kwds(PyObject *mapping)
+{
+    PyObject *newdict = NULL;
+
+    if (mapping != NULL) {
+        assert(PyDict_Check(mapping));
+
+        Py_ssize_t kwlen = PyDict_Size(mapping);
+        assert(kwlen >= 0);
+        if (kwlen > 0) {
+            newdict = _PyDict_NewPresized(kwlen);
+            if (newdict == NULL) {
+                return NULL;
+            }
+            if (_PyDict_MergeEx(newdict, mapping, 2) < 0) {
+                Py_DECREF(newdict);
```

Comment:
Please use PyDict_Copy().

Reply:
`PyDict_Copy` would be incorrect in the general case; since what's being received is only guaranteed to be a `dict` subclass, not necessarily a true `dict`, there are a number of things that can go subtly wrong, and the callee should not need to worry about potentially receiving something that isn't a true `dict` and might behave differently. This pattern of `_PyDict_MergeEx` with `override=2` (which errors on duplicate keys, rather than ignoring or overwriting) is the way other code has been coercing to exact `dict` type for keyword arguments (even when it's just one mapping being coerced; I'm assuming to avoid the risk of a weirdo `dict` subclass that allows and contains duplicate keys).

Reply:
For the record, I agree that it feels weird to reinvent this pattern, but as is, `PyDict_Copy` doesn't quite do the right thing, so the options are make another version of it that lets you specify the `override` mode, or writing off the possibility of `dict` subclasses with repeated keys as being the fault of the author and ignoring them, letting things misbehave silently.

Reply:
OK, then please use `PyDict_New()`, instead of `_PyDict_NewPresized()`.

dict_merge() has fast path for cloning clean dict.

https://github.com/python/cpython/blob/d20bb33f782f5677256d7e1f2462b78113692969/Objects/dictobject.c#L2845-L2856

Reply:
That's reasonable. It'll be a little slower when the fast path can't be used, but I'd expect the typical case to be a freshly allocated `dict`. I wasn't 100% on the correct solution here; seems weird to have a `_PyDict_NewPresized` that's a pessimization in most of the common cases (and since I was checking the size anyway, I figured I'd use it). *shrugs*

Reply:
Change made.